QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java

```diff
@@ -18,25 +18,19 @@
 public class Elevator {
     private static final FloatOutput winch = Igneous.makeJaguarMotor(3, Igneous.MOTOR_FORWARD, 0.4f);
 
-    public static EventOutput setTop;
-    public static EventOutput setMiddle;
-    public static EventOutput setBottom;
-    public static EventOutput stop;
-
     private static final BooleanStatus raising = new BooleanStatus();
     private static final BooleanStatus lowering = new BooleanStatus();
     private static final BooleanStatus goingToMiddle = new BooleanStatus();
 
+    public static EventOutput setTop = EventMixing.combine(BooleanMixing.getSetEvent(raising, true), BooleanMixing.getSetEvent(lowering, false), BooleanMixing.getSetEvent(goingToMiddle, false));
+    public static EventOutput setMiddle = BooleanMixing.getSetEvent(goingToMiddle, true);
+    public static EventOutput setBottom = EventMixing.combine(BooleanMixing.getSetEvent(raising, false), BooleanMixing.getSetEvent(lowering, true), BooleanMixing.getSetEvent(goingToMiddle, false));
+    public static EventOutput stop = BooleanMixing.getSetEvent(BooleanMixing.combine(raising, lowering, goingToMiddle), false);
```

Comment:
This was an issue in the existing code, but should still be fixed: you can probably replace `BooleanMixing.getSetEvent(raising, ...)` with a get from the BooleanStatus - it has an easier-to-use call.